master/buildbot/db/buildsets.py

```diff
@@ -51,7 +51,7 @@ def addBuildset(self, sourcestamps, reason, properties, builderids,
 
         # convert to sourcestamp IDs first, as necessary
         def toSsid(sourcestamp):
-            if isinstance(sourcestamp, type(1)):
+            if isinstance(sourcestamp, type(1)) or isinstance(sourcestamp, type(1L)):
```

Comment:
http://stackoverflow.com/questions/3501382/checking-whether-a-variable-is-an-integer-or-not